### PR TITLE
feat: enable `v2_dev` flag

### PR DESCRIPTION
**Pull request by MichaelDeBoey**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,10 +6,8 @@
     "build": "run-s build:*",
     "build:remix": "remix build",
     "build:server": "esbuild --platform=node --format=cjs ./server.ts --outdir=build --bundle",
-    "dev": "run-p dev:*",
-    "dev:build": "cross-env NODE_ENV=development npm run build:server -- --watch",
-    "dev:remix": "cross-env NODE_ENV=development remix watch",
-    "dev:server": "cross-env NODE_ENV=development node --inspect --require ./node_modules/dotenv/config --require ./mocks ./build/server.js",
+    "dev": "remix dev -c \"npm run dev:serve\"",
+    "dev:serve": "binode --require ./mocks -- @remix-run/serve:remix-serve ./build/server.js",
```
Comment (pcattori) [inline package.json:10]: This stack directly runs the server with `node` without needing `remix-serve`:

```suggestion
    "dev:serve": "node --require ./mocks ./build/server.js",
```

**Review by pcattori — CHANGES_REQUESTED** [review]
Also, looks like typechecking is failing on top-level `await` in `server.ts`

File: package.json [hunk 2]
```diff
@@ -6,10 +6,8 @@
     "build": "run-s build:*",
     "build:remix": "remix build",
     "build:server": "esbuild --platform=node --format=cjs ./server.ts --outdir=build --bundle",
```
Comment (MichaelDeBoey) [inline package.json:8]: @pcattori Do we still need to build the server like this?
I guess we mainly did it to be used in the `dev` script 🤔

File: server.ts [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
 import compression from "compression";
+import type { RequestHandler } from "express";
 import express from "express";
 import morgan from "morgan";
 
+installGlobals();
+
+const BUILD_PATH = path.join(process.cwd(), "build", "index.js");
+/** @type { import('@remix-run/node').ServerBuild | Promise<import('@remix-run/node').ServerBuild> } */
+let build = await import(BUILD_PATH);
```
Comment (MichaelDeBoey) [inline server.ts:17]: Top level await is failing with following TS error


@pcattori Not sure what the right way would be as we're still officially supporting Node 14

File: package.json (hunk 2 above)
Comment (pcattori) [reply package.json]: still needed as this repo authors the custom server in TS and does not use `remix.config.js`'s `server` field to compile the custom server.

File: server.ts (hunk 3 above)
Comment (pcattori) [reply server.ts]: for now I just wrapped this in a `async function run() {...}`

File: package.json (hunk 2 above)
Comment (MichaelDeBoey) [reply package.json]: Should we use `remix.config.js` instead?

**Pull request merged**